apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/RequestResponsePopulateMdcFilter.java

```diff
@@ -52,15 +52,16 @@ public class RequestResponsePopulateMdcFilter extends OncePerRequestFilter {
   private static final String RESPONSE_PREFIX = "response";
 
   /** obfuscated replacement for MBI. */
-  private static final String OBFUSCATED_MBI = "us-mbi|*********";
-
-  /** Regex Pattern to check for an /Patient request end-point. */
-  private static final Pattern PATIENT_REQUEST_REGEX = Pattern.compile("fhir/Patient");
+  private static final String OBFUSCATED_MBI = "*********";
 
   /** Regex Pattern to check for an MBI. */
   private static final Pattern MBI_REGEX =
-      Pattern.compile(
-          "us-mbi\\|[1-9][^SLOIBZsloibz0-9][^SLOIBZsloibz][0-9]-?[^SLOIBZsloibz0-9][^SLOIBZsloibz][0-9]-?[^SLOIBZsloibz0-9][^SLOIBZsloibz0-9][0-9][0-9]");
+      // Real MBIs disallow certain characters, but we'll be conservative here for simplicity's
+      // sake, and so we can easily test against synthetic MBIs.
+
+      // See the following link for more info on the MBI format
+      // https://www.cms.gov/medicare/new-medicare-card/understanding-the-mbi-with-format.pdf
+      Pattern.compile("\\d\\p{Alpha}\\p{Alnum}\\d-?\\p{Alpha}\\p{Alnum}\\d-?\\p{Alpha}{2}\\d{2}");
```

Comment:
Although we don't support MBIs formatted with dashes, the original regex did check for it, and per the document linked above, it is a valid format. So it's plausible to think someone may attempt to construct a query with a dash-formatted MBI.

Reply:
Casting a wider net for potential matches here makes a lot of sense. The scope is already limited by what should or _could_ appear as part of the request. And I agree–it doesn't much imagination to consider how an errant, dash-delimited MBI might be introduced here.